### PR TITLE
Reduce memory allocation on client side because of ResteasyProviderFactory#initialize

### PR DESCRIPTION
**Pull request by asoldano**
Revisiting ResteasyProviderFactory implementations to reduce memory usage on client side.
The changes reduce the number of invocations to ResteasyProviderFactory#initialize() of 70% when running the current RESTEasy testsuite.

### CODE REVIEW

File: resteasy-core/src/main/java/org/jboss/resteasy/core/ResteasyProviderFactoryImpl.java [hunk 1]
```diff
@@ -2361,6 +2367,7 @@ public void injectProperties(Object obj, HttpRequest request, HttpResponse respo
    }
 
    // Configurable
+   //TODO!! problem with listeners...
```
Comment (marekkopecky) [inline resteasy-core/src/main/java/org/jboss/resteasy/core/ResteasyProviderFactoryImpl.java]: Which problem?

Comment (asoldano) [reply resteasy-core/src/main/java/org/jboss/resteasy/core/ResteasyProviderFactoryImpl.java]: uh, stale comment. Basically, as the method returns a live collection, invocations to such method are to be treated the same as the methods that potentially change the ResteasyProviderFactory. That's already the case, though, in DelegateLazyResteasyProviderFactory (I didn't have that class yet when I wrote the comment)

**Comment by marekkopecky** [comment]
@asoldano Do you have a RESTEasy jira for this PR?

**Comment by asoldano** [comment]
@marekkopecky not yet. I need to backport this to 3.6, have the performance team try it and then if it's worth keeping the changes I'll create a jira and merge the PRs

**Comment by marekkopecky** [comment]
@asoldano +1

**Comment by asoldano** [comment]
Closing, a different approach (#1879) is now being evaluated.

**Pull request closed without merging**